src/components/Paginate.tsx

```diff
@@ -0,0 +1,202 @@
+                    currentPage * limit -
+                    (limit - 1)
+                  : currentPage * limit}{" "}
+              </span>
+              of
+              <span className="font-medium"> {totalResults} </span>
+              results
+            </p>
+          </div>
+          <div>
+            <nav
+              className="relative z-0 inline-flex shadow-sm -space-x-px"
+              aria-label="Pagination"
+            >
+              {currentPage !== 1 && (
+                <button // Only show the previous page button if we're not on the first page.
+                  className="relative inline-flex items-center px-2 py-2 rounded-l-md border border-gray-300 bg-primary text-sm font-medium text-gray-500 hover:bg-secondary hover:border-secondary hover:text-primary transition duration-150 ease-in-out"
+                  onClick={() => setCurrentPage(currentPage - 1)}
+                >
+                  <span className="sr-only">Previous</span>
+                  {/* <!-- Heroicon name: chevron-left --> */}
+                  <svg
+                    className="h-5 w-5"
+                    xmlns="http://www.w3.org/2000/svg"
+                    viewBox="0 0 20 20"
+                    fill="currentColor"
+                    aria-hidden="true"
+                  >
+                    <path
+                      fillRule="evenodd"
+                      d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
+                      clipRule="evenodd"
+                    />
+                  </svg>
+                </button>
+              )}
+              <button
+                onClick={() => setCurrentPage(1)}
+                data-e2e-id="first-page-button"
+                className={`border duration-150 ease-in-out font-medium hover:bg-secondary hover:border-secondary hover:text-primary inline-flex items-center px-4 py-2 relative text-sm transition ${
+                  currentPage === 1 // Are we on the first page? If so, make the lefthand size of
+                    ? "bg-secondary border-secondary rounded-l-md text-primary" // the first page's button rounded.
+                    : "bg-primary text-secondary"
+                }`}
+              >
+                1
+              </button>
+              <button
+                onClick={() => {
+                  setCurrentPage(2);
+                }}
+                className={`border duration-150 ease-in-out font-medium hover:bg-secondary hover:border-secondary hover:text-primary inline-flex items-center px-4 py-2 relative text-sm transition ${
+                  currentPage === 2
+                    ? "bg-secondary border-secondary text-primary"
+                    : "bg-primary text-secondary"
+                }`}
+              >
+                2
+              </button>
+              <button
+                onClick={() => setCurrentPage(3)}
```

Comment:
There seems like there's a lot of duplication between these buttons. It would be good to improve the reusability here

Reply:
We can probably accomplish that reusability using a `for` loop.

Some of the pages you linked had that implementation, but I've been so focused on completing the tasks just in this pull request that I haven't had enough time to give every link a thorough viewing.

Reply:
Using a for loop is the right idea! But even better, we can use map here

Reply:
Some suggestions here:
```
// an array of numbers from 1-3
const pages = new Array(3).fill(0).map((e, i) => i + 1)
// then something like this
pages.map(page => (
  <button .../>
));
```

Reply:
I just saw this. I think I tried a different approach, but one that has the same central idea, in e2560fe6608f5c37930fe82d1f23750382d9edb9.

I think it might be more inconvenient to find this conversation to discuss the changes in that commit, so I'll mark this conversation resolved, but we can still comment on the changes in that commit at the commit, where it'll be more easily viewable at the bottom of the conversation tab.